server.go

```diff
@@ -481,14 +487,62 @@ func (srv *Server) Shutdown() error {
 		srv.lock.Unlock()
 		return &Error{err: "server not started"}
 	}
+	// inform the server is no more alive
+	// it will prevent to start read new packet
+	// however the already started READ session will continue.
+	// it can take to MAX(readTimeout, TCPIdleTimeout) to have all READ session processed
 	srv.started = false
 	srv.lock.Unlock()
 
-	if srv.PacketConn != nil {
-		srv.PacketConn.Close()
-	}
+	// and then stop the service : it will wait all queries are served
+
+	timeout := srv.getReadTimeout()
+	idleTimeout := time.Duration(0)
+
+	// we can close the TCP listener (the accept() part)
 	if srv.Listener != nil {
 		srv.Listener.Close()
+
+		// if TCP listener is in use, the read timeout may be extended with the idleTimeout
+		idleTimeout = tcpIdleTimeout
+		if srv.IdleTimeout != nil {
+			idleTimeout = srv.IdleTimeout()
+		}
+	}
+
+	if idleTimeout > timeout {
+		timeout = idleTimeout
+	}
+	timeout = timeout + time.Second
+	if srv.ShutdownTimeout != time.Duration(0) {
+		timeout  = srv.ShutdownTimeout
+	}
+
+	// and then wait all TCPConn and PacketConn finish the on-going work
+	if runtime.GOOS != "windows" {
```

Comment:
don't really care this doesnt(?) work on windows. There is no runtime.GOOS in this package, would not like to see it added

Reply:
ok. I will remove.